fabric-data-attachment-api-v1/src/main/java/net/fabricmc/fabric/mixin/attachment/BlockEntityMixin.java

```diff
@@ -45,4 +71,48 @@ private void readBlockEntityAttachments(NbtCompound nbt, RegistryWrapper.Wrapper
 	private void writeBlockEntityAttachments(RegistryWrapper.WrapperLookup wrapperLookup, CallbackInfoReturnable<NbtCompound> cir) {
 		this.fabric_writeAttachmentsToNbt(cir.getReturnValue(), wrapperLookup);
 	}
+
+	@Override
+	public void fabric_markChanged(AttachmentType<?> type) {
+		this.markDirty();
+	}
+
+	@Override
+	public void fabric_acknowledgeSyncedEntry(AttachmentType<?> type, @Nullable Object value) {
+		if (this.hasWorld() && !this.world.isClient) {
```

Comment:
use `isClient()` method instead of the `isClient` field (there are many instances of this)